nullaway/src/test/java/com/uber/nullaway/NullAwayKeySetIteratorTests.java

```diff
@@ -0,0 +1,132 @@
+package com.uber.nullaway;
+
+import org.junit.Test;
+
+public class NullAwayKeySetIteratorTests extends NullAwayTestsBase {
+
+  @Test
+  public void mapKeySetIteratorBasic() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import java.util.*;",
+            "public class Test {",
+            "  public void keySetStuff(Map<Object, Object> m) {",
+            "    for (Object k: m.keySet()) {",
+            "      m.get(k).toString();",
+            "    }",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void mapKeySetIteratorShadowing() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import java.util.*;",
+            "public class Test {",
+            "  private final Object k = new Object();",
+            "  public void keySetStuff(Map<Object, Object> m) {",
+            "    for (Object k: m.keySet()) {",
+            "      m.get(k).toString();",
+            "    }",
+            "    // BUG: Diagnostic contains: dereferenced expression m.get(k) is @Nullable",
+            "    m.get(k).toString();",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void mapKeySetIteratorDeeperAccessPath() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import java.util.*;",
```

Comment:
Even here, I'd just import `Map` and `HashMap`.

Reply:
Just, in general, mild vote for listing the imports individually as if this were real code. But I am open to the opposite argument.

Reply:
Agreed, fixed in 4257191